zerver/lib/narrow.py

```diff
@@ -161,7 +161,10 @@ def is_spectator_compatible(narrow: Iterable[NarrowParameter]) -> bool:
     ]
     for element in narrow:
         operator = element.operator
-        if operator not in supported_operators:
+        operand = element.operand
+        if (operator not in supported_operators) and not (
+            operator == "is" and operand == "resolved"
+        ):
             return False
```

Comment:
This is confusingly written; just add a `if (condition): continue`  block above the one that does `return False`, rather than making this hard-to-read expression.